addon/src/index.js

```diff
@@ -1,7 +1,21 @@
-import { assert, deprecate } from '@ember/debug';
+import { assert } from '@ember/debug';
 import { get } from '@ember/object';
 import { tracked } from '@glimmer/tracking';
 import { createCache, getValue } from '@glimmer/tracking/primitives/cache';
+import {
+  macroCondition,
+  dependencySatisfies,
+  importSync,
+} from '@embroider/macros';
+
+let deprecate;
+
+if (macroCondition(dependencySatisfies('ember-source', '^3.28.0 || ^4.0.0'))) {
+  deprecate = importSync('@ember/debug').deprecate;
+} else {
+  // eslint-disable-next-line no-undef
+  deprecate = Ember.deprecate;
```

Comment:
this is ugly, *but* `importSync('@ember/application/deprecations')` does not work.
I assume it fails because in Ember.js <= 3.25 such module does not exist and it's just build-time replacement with global `Ember` via babel plugin.

IMO we should remove the deprecated functionality in 2.0.0 anyways so follow up PR will remove need for the `deprecate` function.

Reply:
Seems good. Also, per Ed's comment on a *different* PR, we can just drop support for 3.24 since it's out of LTS.

Reply:
@chriskrycho makes sense, although I don't think we *have to* explicitly drop 3.24 even though it's out of LTS as it does not cost us anything in the runtime.
but that's probably another discussion for how long backwards we should strive to support (spoiler - if no extra polyfills/much code I would try to go back as much as feasible)